misc/python/materialize/parallel_workload/executor.py

```diff
@@ -39,8 +40,12 @@ class Executor:
     rollback_next: bool
     last_log: str
     action_run_since_last_commit_rollback: bool
+    _atomic_counter: int
```

Comment:
I'm a little confused by these variables -- are these all meant to be class variables shared by all instances of the class? Or instance variables that are defined in the `__init__` method? This seems like you are doing both but shadowing all class variables with individual instance variables?

Reply:
Yes, you are right. I will remove mine.

I will let the others up to @def-.

Reply:
In Python it's slightly messy: https://mypy.readthedocs.io/en/stable/class_basics.html#instance-and-class-attributes

> As in Python generally, a variable defined in the class body can be used as a class or an instance variable.

But I still consider this to be the nicest way to have instance variable type definitions, because putting them in each.